docs/content/developer/workshops/core/IV-capstone-dapp-kit.mdx

```diff
@@ -0,0 +1,299 @@
+    <div style={{ padding: 16 }}>
+      <ConnectButton />
+      {account && (
+        <>
+          <div style={{ marginTop: 12 }}>Connected: {account.address}</div>
+          <OwnedObjects />
+          <RequestGrant />
+        </>
+      )}
+    </div>
+  );
+}
+```
+
+## Approve a request from the CLI
+
+An issuer (admin-gated) should approve a student’s request by calling `lifecycle::approve_grant` from the CLI. This mints and transfers the Grant to the student by calling into the `grant` module.
+
+You need the following IDs:
+- `PKG_ID`: the published package ID
+- `REQUESTS_ID`: the shared Requests object ID
+- `ROLES_ID`: the shared Roles object ID (from Workshop III)
+- `ADMIN_CAP_ID`: the AdminCap object ID held by an issuer/admin
+- `STUDENT_ADDR`: the student’s address that submitted the request
+
+Example using the CLI:
+
+```
+iota client call \
+  --package <PACKAGE-ID> \
+  --module lifecycle \
+  --function approve_grant \
+  --args <REQUESTS_ID> <STUDENT_ADDR> <ROLES_ID> <ADMIN_CAP_ID> \
+  --gas-budget 80000000
+```
+
+Notes:
+- Double-check argument order: `approve_grant(requests, student, roles, cap, ctx)`.
+- The function reads and removes the student’s pending amount from `Requests`, then mints and transfers a `Grant` to the student address.
+
+:::tip Open discussion
+How are we still indexing this grant if we approve it this way?
+:::
+
+## See your grant in the dApp
+
+After a successful `approve_grant` call, refresh your dApp. The `OwnedObjects` section should now list the new `Grant` object owned by the student’s address.
+
+## Optional: Admin panel actions
+
+If you plan to expose admin-only operations (e.g., creating the shared Roles object or indexing grants), gate them in the UI by checking that the connected account is allowed (for example, show the admin panel only for a known publisher address). These calls typically require passing capability object IDs in the move call, which you should store in a config file and never hardcode in public deployments.
+
+- Example admin calls (adjust to your modules):
+  - access::create_roles(&AdminCap)
+  - registry::index_grant(&mut Registry, student, grant_id, &AdminCap)
+
+Reuse the `useSignAndExecuteTransaction` pattern from RequestGrant.
+
+## Run the app
+
+```bash
+npm run dev
+```
+
+Open the local URL and connect a wallet to try reading owned objects and submitting a request. After an issuer approves your request from the CLI, refresh the page to see the minted Grant among your owned objects.
+
+## Next steps
+
+- Add a dashboard for admins/issuers:
+  - List pending requests from the shared `Requests` object.
+  - Approve/deny requests with a single click (backend via dApp Kit calls gated by wallet address, or CLI).
+- Improve the UI:
+  - Replace inline styles with a component library (e.g., Tailwind, Mantine, MUI).
+  - Add success/error toasts, spinners, and empty states.
+  - Surface on-chain events (e.g., `GrantRequested`, `GrantApproved`) in the UI.
```

Comment:
Do we want to also add " call to actions" in the end of this four series core workshop?

> Need help or want to collaborate? Join the [IOTA Builders Discord](https://discord.gg/iota-builders) to share progress, get feedback on economic models, and solve technical challenges with the community. Share your factory designs and creator dashboards!

Like this